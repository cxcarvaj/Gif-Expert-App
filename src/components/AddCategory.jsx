import { useState } from 'react';

export const AddCategory = ({ categories, onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onSubmit = ( event ) => {
        event.preventDefault();

        const newInputValue = inputValue.trim();

        if ( newInputValue.length <= 1 ) return;

        // setCategories( (categories) => [inputValue, ...categories] );
        onNewCategory( newInputValue );
        setInputValue('');
    }

    const handleInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type = 'text'
                placeholder = 'Search a Gif'
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    )

};