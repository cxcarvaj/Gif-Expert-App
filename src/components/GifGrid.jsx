import PropTypes from 'prop-types';

import { GifGridItem } from './index';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3> { category } </h3>

            { isLoading && <p>Loading...</p> }

            <div className='card-grid'>

                { 
                    images.map( ( image ) => (
                        <GifGridItem 
                            key = {image.id} 
                            // title = {title} 
                            // url = {url}
                            { ... image }
                        />
                    ))
                }

            </div>
        </>
    )

};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}