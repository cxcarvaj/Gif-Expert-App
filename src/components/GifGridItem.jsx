export const GifGridItem = ( { title, url } ) => {

    return (
        <>
            <div className='card'>
                <img src = { url } alt = { title } />
                <p>{ title }</p>
            </div>
        </>
    )

};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string,
};