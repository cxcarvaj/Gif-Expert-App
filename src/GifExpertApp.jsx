import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['The Office']);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) {

            return alert(`Category: ${newCategory} already exists!`)
        };
        setCategories([newCategory, ...categories]);
        // setCategories( (categories) => [...categories, 'New Game'] );
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onNewCategory={onAddCategory}
            />

            {
                categories.map((category) => (
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                ))
            }

        </>
    );
};
