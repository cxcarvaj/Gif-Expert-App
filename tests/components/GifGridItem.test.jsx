import { GifGridItem } from '../../src/components';
import { render, screen } from '@testing-library/react';

describe("Tests in <GifGrid />", () => {

    const title = 'The Office';
    const url = 'https://media.giphy.com/media/3o7TKsQ8UQY1NlC5aU/giphy.gif';

    test('should match with the snapshot', () => {


        const { container } = render( <GifGridItem
                                        title = { title }
                                        url = { url }
                                    />);

        expect( container ).toMatchSnapshot();

    });
    

    test('should show the image with the URL and the ALT text provided', () => {


        const { container } = render( <GifGridItem
                                        title = { title }
                                        url = { url }
                                    />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('should show the title in the component', () => {

        

        const { container } = render( <GifGridItem
                                        title = { title }
                                        url = { url }
                                    />);

    
        const screenTitle = screen.getByText( title );

        expect( screenTitle ).toBeTruthy();

    });
});