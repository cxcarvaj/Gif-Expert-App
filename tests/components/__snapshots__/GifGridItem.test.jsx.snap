// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The Office"
      src="https://media.giphy.com/media/3o7TKsQ8UQY1NlC5aU/giphy.gif"
    />
    <p>
      The Office
    </p>
  </div>
</div>
`;
